samples/todo-android/app/src/main/java/com/squareup/sample/todo/TodoEditorWorkflow.kt

```diff
@@ -0,0 +1,118 @@
+@file:OptIn(WorkflowUiExperimentalApi::class)
+
+package com.squareup.sample.todo
+
+import com.squareup.sample.todo.TodoEditorOutput.Done
+import com.squareup.sample.todo.TodoEditorOutput.ListUpdated
+import com.squareup.workflow1.Snapshot
+import com.squareup.workflow1.StatefulWorkflow
+import com.squareup.workflow1.action
+import com.squareup.workflow1.asWorker
+import com.squareup.workflow1.runningWorker
+import com.squareup.workflow1.ui.TextController
+import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
+
+sealed class TodoEditorOutput {
+  data class ListUpdated(val newList: TodoList) : TodoEditorOutput()
+  object Done : TodoEditorOutput()
+}
+
+/**
+ * Renders a given a [TodoList] as a [TodoEditorScreen]. Emits updated copies
+ * of the list as output, via [ListUpdated]; or [Done] to indicate that editing
+ * is complete.
+ *
+ * Note that a running instance ignores changes to its props, which works fine
+ * in this sample but would be inadequate in real life. To change that,
+ * add an implementation of [onPropsChanged].
+ */
+class TodoEditorWorkflow :
+  StatefulWorkflow<TodoList, TodoEditingSession, TodoEditorOutput, TodoEditorScreen>() {
+  override fun initialState(
+    props: TodoList,
+    snapshot: Snapshot?
+  ): TodoEditingSession = props.toEditingSession()
+
+  override fun snapshotState(state: TodoEditingSession): Snapshot? = null
+
+  override fun render(
+    renderProps: TodoList,
+    renderState: TodoEditingSession,
+    context: RenderContext
+  ): TodoEditorScreen {
+    // Monitor the title and each row for text changes.
+    context.runningWorker(renderState.title.onTextChanged.asWorker(), "title") { textChanged }
```

Comment:
nit:
```suggestion
    context.runningWorker(renderState.title.onTextChanged.asWorker(), "title", textChanged)
```

I actually don't know whether the compiler optimizes anything here in terms of callstack but any reason why we can't avoid another anonymous lambda wrapper?

Reply:
Worker requires a function returning an action, can't pass the action directly.

Reply:
Maybe ::textChanged, but that's just a cosmetic change.

Reply:
Yeah, no, that doesn't work. This is just how it is.

Reply:
ah of course. I think I always blur one of the functional layers between the *handler* and the *action* that it returns (which is also a function). Could we overload `runningWorker` to also just take an action function in the cases where the handler doesn't have any side effects/work to do?

Reply:
That's a pretty rare case, though. I don't think we want to simplify the ability to ignore the worker's output.

Reply:
Ack. Ok that makes sense.